versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -250,6 +250,15 @@ protected CommitLogEntry commitAttempt(
     Function<Hash, CommitLogEntry> currentCommit =
         h -> h.equals(branchHead) ? currentBranchEntry : null;
 
+    for (KeyWithBytes put : commitParams.getPuts()) {
+      if (put.getPayload() <= (byte) 0) {
```

Comment:
Do we actually expect this condition to be `true`? We validate payload bytes to be > 0 at the type level, do we not?

Reply:
We do - probably overly cautious here.